### PR TITLE
Preprocess analysis

### PR DESCRIPTION
**Pull request by josephignace**
Preprocess and data cleaning scripts updated (see GI and MH). 
Create project actions updated (preprocess, data cleaning, and list of actions).
Active_analyses script updated: list of covariates updated for each model, new analyses added (prior history, parkinson risk, vascular risk, prior cognitive impairment, prior parkinson).

### CODE REVIEW

File: analysis/preprocess/Stage1_data_cleaning.R
Comment (venexia) [inline analysis/preprocess/Stage1_data_cleaning.R:393]: The code as written is removing outcome events but not excluding individuals. We want to exclude these individuals altogether.

As above, I think this condition is best applied at the make_model_input stage as otherwise we have to maintain this Parkison's disease cohort throughout the pipeline along with the other outcome specific cohorts.

File: analysis/create_project_actions.R
Comment (venexia) [inline analysis/create_project_actions.R:156]: [Optional] We only need the describe_* actions if something goes wrong so these could be removed.

**Review by venexia — CHANGES_REQUESTED** [review]
Hi @josephignace. Thank you for submitting this PR. There are some changes that need to be addressed - these include the following:
- The active analyses table contains subgroup analyses for outcomes that are not specified in the protocol
- There is decision to be taken about which outcomes to run the core subgroups (e.g., age, sex, ethnicity) for [we can discuss this at out meeting on Tuesday]
- The coding of the high vascular risk subgroup in the preprocessing script needs revising as it generates missing values
- The neuro specific exclusion criteria are currently misspecified and I think should be implemented later in the pipeline
- Naming conventions for rounded variables have not been implemented

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:49]: I will wait for the meeting. Also, the specific analysis for neuro should be included in the meeting.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:410]: Models removed.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:498]: Changes added. Wait for the meeting to review the models.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:476]: Changes added. Wait for the meeting to review the models.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:564]: Changes added. Wait for the meeting to review the models.

File: analysis/preprocess/preprocess_data.R
Comment (josephignace) [reply analysis/preprocess/preprocess_data.R:100]: Removed.

File: analysis/preprocess/preprocess_data.R
Comment (josephignace) [reply analysis/preprocess/preprocess_data.R:130]: Variable renamed. Which ones should be TRUE and which FALSE? Message is not clear.

File: analysis/preprocess/preprocess_data.R
Comment (josephignace) [reply analysis/preprocess/preprocess_data.R:145]: Removed.

File: project.yaml
Comment (josephignace) [reply project.yaml:1]: Step used to select which outcomes will run all or main analyses, based on the number of patients per outcomes.

File: analysis/preprocess/preprocess_data.R
Comment (josephignace) [reply analysis/preprocess/preprocess_data.R:1]: Removed.

File: analysis/preprocess/Stage1_data_cleaning.R
Comment (josephignace) [reply analysis/preprocess/Stage1_data_cleaning.R:418]: This is not clear. Neither MH nor GI have the rounded detail you mention.

File: analysis/preprocess/Stage1_data_cleaning.R
Comment (josephignace) [reply analysis/preprocess/Stage1_data_cleaning.R:368]: Removed.

File: analysis/create_project_actions.R
Comment (josephignace) [reply analysis/create_project_actions.R:156]: It does not affect leaving such actions then.

**Review by venexia — CHANGES_REQUESTED** [review]
Hi @josephignace. Thank you for the further work on this pull request. There remain some comments that need to be addressed before I can approve this for merging. Please find my replies on the relevant comments and let me know if anything is unclear.

File: analysis/preprocess/Stage1_data_cleaning.R
Comment (josephignace) [reply analysis/preprocess/Stage1_data_cleaning.R:390]: Added.

File: analysis/preprocess/preprocess_data.R
Comment (josephignace) [reply analysis/preprocess/preprocess_data.R:135]: None of you messages is clear, it requires a lot of guessing. Communication is not effective and not clear.

File: analysis/preprocess/preprocess_data.R
Comment (josephignace) [reply analysis/preprocess/preprocess_data.R:99]: Removed.

File: analysis/preprocess/preprocess_data.R
Comment (josephignace) [reply analysis/preprocess/preprocess_data.R]: Moved to BMI (create obesity).

File: analysis/preprocess/Stage1_data_cleaning.R
Comment (venexia) [reply analysis/preprocess/Stage1_data_cleaning.R:380]: The variable `removed` is derived from a midpoint6 rounded variable and should have the prefix "_derived" as per OpenSAFELY documentation: https://docs.opensafely.org/releasing-files/#naming-convention-for-midpoint-6-rounding

**Review by venexia — CHANGES_REQUESTED** [review]
Hi @josephignace. Thank you for further revising this pull request. There is just one comment regarding variable names to be addressed and then I can approve this request.

**Comment by josephignace** [comment]
> Hi @venexia. Variable has been renamed.

**Review by venexia — APPROVED** [review]

**Pull request merged**
